client/me/purchases/manage-purchase/index.tsx

```diff
@@ -639,6 +642,76 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! purchase?.isRefundable || ! purchase?.refundText || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase?.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase?.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		const refundText = (() => {
+			// User would receive another free domain credit upon cancelling.
+			if ( isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
+			  return translate(
+				"If you're unsure about this domain, cancel it before {{refundDate/}} to choose another free domain instead.",
+				{
+					components: {
+						refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+					},
+				}
+			);
+			}
+		  
+			// User would receive a financial refund upon cancelling.
+			return translate(
+				'Eligible for a refund of %(refundAmount)s until {{refundDate/}}.',
+				{
+					components: {
+						refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+					},
+					args: {
+						refundAmount: purchase?.refundText,
+					},
+					context: 'refundAmount contains the currency and amount - eg. £20',
+				}
+			);
+		  }) ();
+
+
+		// Neither a financial refund nor another domain credit would be available.
+		if ( ! isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
+			return;
+		}
+
+		return (
+			<div className="manage-purchase__refund-text-wrapper">
+				<span className="manage-purchase__refund-text">{ refundText }</span>
+				<InfoPopover iconSize={ 16 } position="top left">
+					<span>
+						{ translate(
+							'The refund period for this purchase is %(refundPeriodInDays)s days. Questions? Check our {{link}}refund policy{{/link}}.',
+							{
+								components: {
+									link: <InlineSupportLink supportContext="refund-periods" showIcon={ false } />,
+								},
+								args: {
+									refundPeriodInDays: purchase?.refundPeriodInDays,
```

Comment:
Same here. `refundPeriodInDays` is typed as a number but this allows it to be undefined, so that if the code above here changed in the future to allow a falsy `purchase` object through, this would happily print `The refund period for this purchase is undefined days`, which we can prevent by letting TypeScript guard against it:

```suggestion
									refundPeriodInDays: purchase.refundPeriodInDays,
```